src/config/output.cpp

```diff
@@ -35,14 +35,14 @@ namespace config {
 
 using namespace boost::program_options;
 
-output::output()
+output::output(const bc::settings& settings)
```

Comment:
Losing this default constructor turns out to be a problem. In `bc::explorer::commands::tx-encode` this is used for `boost::program_options` which in turn uses `boost::lexical_cast` internally. `boost::lexical_cast` requires the target object to be default-constructible.

Reply:
The config classes should all allow default construction. In this case it's acceptable to just remove the amount validation. In the case of header (and therefore block) it appears that we could eliminate the need for settings by parameterizing:
```cpp
    bool is_valid_timestamp() const;
    bool is_valid_proof_of_work(bool retarget) const;

    code check(bool retarget) const;
```

Though I'm not sure about the downstream impact. I suppose this brings us back to your previous suggestion.